test/modules/tilesource-dynamic-url.js

```diff
@@ -119,7 +119,8 @@
 
             OpenSeadragon.extend( Tile.prototype, OpenSeadragon.Tile.prototype, {
                 getUrl: function() {
-                    ASSERT.ok(true, 'Tile.getUrl called');
+                    // TODO: it seems sometimes this can be called after ASSERT is destroyed; should this be fixed differently?
```

Comment:
Unfortunately I don't know enough about `ASSERT` to make a suggestion here. Do you know why we're using it instead of `assert`?

Reply:
That was introduced when the entire test was created a couple of years back in https://github.com/openseadragon/openseadragon/commit/601160e9fb4a6a52b95f945fa7d1eccf7c851a5d 

Looks like the `Tile` prototype is modified to do the assertion, and this is configured in `beforeEach` which doesn't have access to `assert`. To get around this, `assert` is bound to a `ASSERT` at a scope that can be accessed at that location.

I'm not sure what the right approach to fixing this is, though I can look into it some more.

Reply:
I see! Maybe we don't need to set `ASSERT` to `null` in the `afterEach`? Or is there some way to make sure all of the tests are done before we get to `afterEach`? There's the async test functionality... Maybe that needs to be used here.

Reply:
> There's the async test functionality... Maybe that needs to be used here.

I'm really not that familiar with qunit. Pointers on how to do this would be helpful. I can also look at the other modules to see if there's a pattern to use.

Reply:
Here's an example of the async functionality: 

https://github.com/openseadragon/openseadragon/blob/6abcc76a93ee3369267908f0a80641dc171440ed/test/modules/ajax-tiles.js#L65

Reply:
Added `assert.async()` to ensure the test waits until all async calls are done before proceeding to `afterEach`.